docs/DevGuideFeatureImplementations/suggest.adoc

```diff
@@ -0,0 +1,50 @@
+// tag::suggest[]
+=== [Proposed] Suggest feature
+
+==== Proposed Implementation
+
+The suggest mechanism is implemented through a `SuggestCommand` that extends from the abstract `Command` class. It interacts with the object that implements the `Model` interface by updating it with dishes that are within the calories budget. The calories budget is obtained from `Model#getRemainingCalories()`.
+
+Additionally, it implements the following operation:
+
+* `Model#getRemainingCalories` -- gets the current allowed calories budget.
+
+This operation is exposed in the `Model` interface as `Model#getRemainingCalories()`.
+
+Given below is an example usage scenario and how the suggest mechanism behaves at each step.
+
+Step 1. The user starts up CaloFit for the first time. The `DishDatabase` will be loaded with the initial data by calling `MainApp#loadDishDatabase`.
+
+[NOTE]
+If the DishDatabase is empty, or the daily calorie budget have exceeded the set amount, or there are no dishes that is within the amount, nothing will be display.
+
+Step 2. The user enters `"suggest"` in the Command Line Input to invoke the `SuggestCommand` command which will run `SuggestCommand#execute()`. `Model#getRemainingCalories()` will be executed to get the remaining calories, which will provide `#Model#updateFilteredDishList` with the calories budget to update the list accordingly.
+
+Step 3. The success message will be return, while printing the updated dish list on CaloFit.
+
+The following sequence diagram shows how the suggest operation works:
```

Comment:
Is `deduct c from remainingBudget` a method call, should it be in a sequence diagram? 

Are the self invocation methods drawn correctly? It doesn't seem clear?